src/dataprotection/azext_dataprotection/manual/custom.py

```diff
@@ -255,6 +260,151 @@ def dataprotection_backup_instance_list_from_resourcegraph(client, datasource_ty
+        data_entity = get_client(cmd.cli_ctx, ResourceType.DATA_KEYVAULT)
+        data_client = data_entity.client_factory(cmd.cli_ctx, None)
+        secrets_list = data_client.get_secrets(vault_base_url=keyvault.properties.vault_uri)
+        given_secret_uri = backup_instance['properties']['datasource_auth_credentials']['secret_store_resource']['uri']
+        given_secret_id = helper.get_secret_params_from_uri(given_secret_uri)['secret_id']
+        valid_secret = False
+        for secret in secrets_list:
+            if given_secret_id == secret.id:
+                valid_secret = True
+                break
+
+        if not valid_secret:
+            raise CLIError("Secret URI provided is not valid")
+
+        if keyvault.properties.public_network_access == 'Disabled':
+            raise CLIError("Keyvault needs to have public network access enabled")
+
+        keyvault_permission_models = manifest['secretStorePermissions']
+        if keyvault.properties.enable_rbac_authorization:
+            keyvault_rbac_model = keyvault_permission_models['rbacModel']
+            role = keyvault_rbac_model['roleDefinitionName']
+
+            keyvault_assignment_scope = helper.truncate_id_using_scope(keyvault_id, permissions_scope)
+
+            role_assignment = list_role_assignments(cmd, assignee=principal_id, role=role, scope=keyvault_id, include_inherited=True)
+            if not role_assignment:
+                role_assignments_arr.append(create_role_assignment(cmd, assignee=principal_id, role=role, scope=keyvault_assignment_scope))
+
+        else:
+            keyvault_permissions_map = {policy.object_id: policy.permissions for policy in keyvault.properties.access_policies}
+            from azure.cli.command_modules.keyvault.custom import set_policy
+            vault_access_policy_model = keyvault_permission_models['vaultAccessPolicyModel']
+            access_policies = vault_access_policy_model['access_policies']
+            vault_permissions = access_policies['permissions']
+
+            vault_secret_permissions = vault_permissions['secrets']
+            secrets_array = keyvault_permissions_map[principal_id].secrets
+
+            permissions_set = True
+            for permission in vault_secret_permissions:
+                if permission not in secrets_array:
+                    permissions_set = False
+                    secrets_array.append(permission)
+
+            if not permissions_set:
+                set_policy(cmd, keyvault_client, keyvault_rg, keyvault_name, object_id=principal_id, secret_permissions=secrets_array)
+
+        from azure.cli.command_modules.keyvault.custom import update_vault_setter
+
+        update = False
+        if keyvault.properties.network_acls:
+            if keyvault.properties.network_acls.bypass == 'None':
+                keyvault.properties.network_acls.bypass = 'AzureServices'
+                update = True
+
+            if update:
```

Comment:
if update is getting set to true only in the above case then it is better to more the update_vault_setter function call to that condition itself and get rid of the update variable.